indexers/providers/flower_power.go

```
@@ -136,33 +149,44 @@ func (f FlowerPower) IndexAllThings(indexContinuously bool) error {
 // does not contain the userID - as in the case of a new user POST request -
 // remote Things will be fetched from their 1st sample. In order to speed up the
 // fetching process parallel requests are executed.
-func (f FlowerPower) IndexThingsForUser(userIdentity types.Identity) ([]thingfulx.Thing, error) {
+func (f FlowerPower) IndexThingsForUser(userIdentity types.Identity, indexAllThings bool) error {
```

i think you said this yesterday, but remind me again why return an error here that we never look at? Seems we should maybe also document this function to make it clear it is intended to be called asynchronously and remove the returned error. 

Actually I have another idea that might be a cleaner approach here, what if we have the `IndexThingsForUser` return all errors without logging like we'd write in normal idiomatic code with errors to propagate back, but then add an `AsyncIndexThingsForUser`function that wraps the synchronous function and does the goroutine spawning, logging and swallowing of the error there?

---

The main reason why I left the error there was to make it easier to test the function and make sure that errors are "captured" properly. I wasn't quite sure how to go about fully testing it if nothing was returned.

I like your idea about having an Async wrapper around `IndexThingsForUser`. It would make `IndexThingsForUser` re usable in a synchronous context if needed.